spv/sync.go

```diff
@@ -569,13 +574,17 @@ func (s *Syncer) breaksMinVersionTarget(rp *p2p.RemotePeer) bool {
 
 func (s *Syncer) connectToPersistent(ctx context.Context, raddr string) error {
 	for {
-		s.connectAndRunPeer(ctx, raddr, true)
-
-		// Retry persistent peer after 5 seconds.
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-time.After(5 * time.Second):
+		default:
+		}
+
+		backoff := s.connectAndRunPeer(ctx, raddr, true)
+		if backoff {
+			// Delay next connect attempt to save resources and not overwhelm peers,
+			// it's unlikely to succeed anyway if we do retry immediately.
+			time.Sleep(5 * time.Second)
```

Comment:
These are persistent peers which I don't guess many people set. I think it can stay simple. It won't spin up the cpu with the sleep that was already there.

sleep and what was already there do the same thing, except one does not block shutdown. Not blocking shutdown is the correct thing to do here.

Reply:
> Not blocking shutdown is the correct thing to do here.

I don't disagree with that, I've adjusted the way `backoff` is returned from `connectAndRunPeer` and now I believe it should never sleep on `ctx` canceled (which is what happen on shutdown for example).

> sleep and what was already there do the same thing, except one does not block shutdown.

We could just ignore `backoff` value in `connectToPersistent` and wait with `time.After` (like it was before), but why not just use `backoff` param with sleep (assuming I've implemented it correctly) just to be uniform with how it is done in `connectToCandidates` now.

We could also just remove `backoff` and do it all with unconditional `time.After` like it was in the past. But that way we'll sleep for no reason when switching from one peer to another, for example.